extensions/jdbc/jdbc-mariadb/deployment/src/main/java/io/quarkus/jdbc/mariadb/deployment/JDBCMariaDBProcessor.java

```diff
@@ -58,4 +58,9 @@ void registerServiceBinding(Capabilities capabilities,
         }
         dbKind.produce(new DefaultDataSourceDbKindBuildItem(DatabaseKind.MARIADB));
     }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    void addNativeImageResources(BuildProducer<NativeImageResourceBuildItem> resources) {
+        resources.produce(new NativeImageResourceBuildItem("mariadb.properties", "driver.properties"));
```

Comment:
Not sure why those weren't added before (they also exist prior to 3.0) and I was seeing a (non-blocking?) NPE w.r.t. to `mariadb.properties`, so I added both.

Reply:
If a property files is A)Totally unused or B)Used only during class initialization, then we don't need to include a copy in the native image, as the constants will already have been loaded from it.

Reply:
it seems `mariadb.properties` is used only by `org.mariadb.jdbc.util.VersionFactory`; Ok to add it this way, going forward we could try to contribute a patch to make this a "real constant", or substitute to fold it into a constant via @Fold - doesn't seem urgent though, perhaps just add a comment about it being used in org.mariadb.jdbc.util.VersionFactory ?

Regarding `driver.properties` ; it's used by `DriverPropertyInfo[] getPropertyInfo(String url, Properties info) throws SQLException`; similarly I think we could do better in the future as no known code is using this method; we should probably register this as a conditional based on this method being reachable. Could you just add a comment? Or I'll follow up later :)

Reply:
W.r.t. `mariadb.properties`, woukd that be a valid thing to do?
```java
@TargetClass(org.mariadb.jdbc.util.VersionFactory.class)
final class VersionFactory_Substitutions {

    @Alias
    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FromAlias, isFinal = true)
    private static Version instance;

    static {
        try (InputStream inputStream = Thread.currentThread().getContextClassLoader()
                .getResourceAsStream("mariadb.properties")) {
            Properties prop = new Properties();
            prop.load(inputStream);
            instance = new Version(prop.getProperty("version"));
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }

    @Substitute
    public static Version getInstance() {
        return instance;
    }
}
```
(especially that `static` initializer)

> or substitute to fold it into a constant via @fold

Can you eloborate more on this? I couldn't find any `@Fold` example (or even the annotation).

W.r.t. `driver-properties`:
> we should probably register this as a conditional based on this method being reachable.

Can you point me to an example or docs?

Reply:
I removed `driver.properties` because it's only used for (optional) descriptions of driver config params and that entire methode is probably never called in this Quarkus setup. Less is more.

Regarding `mariadb.properties`, the question from above remains whether that `static` block is "legal" (it does get the job done though).

Reply:
Frankly, if the file is not huge, I would just include it into the native executable instead of complexifying things.

Reply:
I don't mind really. WDYT @Sanne?

Reply:
I don't mind either. I like your substitution in principle but agree we shouldn't have substitutions when we can avoid them... my suggestion was more hypotethical about contributing a similar patch upstream: if the driver could store the `Version` initialized in a static block, we wouldn't need patching. But I wonder if there's a reason they went to such lenghts of lazily initializing it (?).

Regarding the `@Fold` annotation... it seems it has been moved in GraalVM into an internal module - but don't worry it's not important. In principle the Fold annotation allows (allowed?) a method computation to be replaced with a constant, whose value is defined by running that method once during build time.

Reply:
> But I wonder if there's a reason they went to such lenghts of lazily initializing it (?).

Yeah, no idea. I also wondered why they didn't just leverage `MANIFEST.MF` plus `java.lang.Package.getImplementationVersion()`.

Ok, so I'll remove that substitution again and add back the inclusion of `mariadb.properties`.
I'll also add a note why `driver.properties` isn't included. Then we should be good to go/merge!